src/decode.rs

```diff
@@ -215,15 +214,22 @@ impl<S: Stream> Decoded<S> {
                 match line.last().unwrap() {
                     b'\r' => {
                         incomplete_line.extend_from_slice(&line[..line.len() - 1]);
+                        let il = this.incomplete_line.take();
```

Comment:
This fixed an implementation issue with my other PR so already the SSE test harness is worth it. :)